### PR TITLE
Airlock Overlays + Tweaks

### PR DESCRIPTION
**Pull request by uraniummeltdown**
airlocks

From /tg/:
This changes the way airlock icons work, from flick()'ing whole sprites to overlaying images. All elements have been separated so it's easier to sprite new airlocks.

Improved airlock assembly construction, examining an assembly gives you hints and mineral airlocks can finally have glass added to them. Assembly construction gives more feedback to the user
Station2/Glass airlocks (the ones that are in a row of 3 in hallways) now have a solid version, as do survival pod and multi-tile airlocks.
Multi-tile airlocks are now made with 8 metal because it's a double airlock
Added abductor airlocks, assembly made with 4 alien alloy
Added external maintenance airlocks, a darker maint airlock
Added more airlocks to the RCD
Glass maintenance and glass external airlocks are here.
Cult unruned and runed airlocks finally have different sprites (person who ported them used same sprite for both), the cult overlay actually shows up now
Some old airlock icons kept in due to use as /door/unpowered by away missions (like the hotel)
Airlocks open faster
Added isshade() and isconstruct() helpers
The map edits are find'n'replace and correcting stupid var edits
Made glass airlocks into a subtype of their regular variant instead of a whole other airlock
door assembly types moved to their own file
p_open -> panel_open
Added attaching paper/photos to airlocks

:cl:
add: Examining an airlock assembly gives construction hints and shows what pen name is set. Mineral airlocks can have glass added to them.
add: Added public and external maintenance airlock assemblies to metal recipes
add: Maintenance, standard and external airlocks now have glass versions. Multi-tile airlock has a solid version.
add: Added more airlocks to RCD
add: You can attach paper and photos to airlocks, wirecutters to remove them
tweak: Airlocks open faster
tweak: Vault and high-security airlocks are made with 6 plasteel up from 4
fix: Plasteel should have the correct amount of materials in it now
/:cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
Pics plz

**Comment by taukausanake** [comment]
_*scream *scream *scream_

I don't think I like this. It's true _maybe_ having each layer of an airlock sprite separate so you can make new sprites is good, but last I think /tg/ had that saved in a photoshop file with all the layers. Having them separate in a .dmi will be disgusting. Maybe I don't know what I'm talking about, but just in case I do and the goal is to make sprites easier to edit and create, I recommend adding a Paint.NET or similar program with layer painting file to the files

**Comment by uraniummeltdown** [comment]
@taukausanake There already is a .psd file for airlocks, icons/PSD files/door.psd

**Comment by uraniummeltdown** [comment]
This is ready for review

File: code/game/atoms.dm [hunk 1]
```diff
@@ -613,6 +613,9 @@ var/list/blood_splatter_icons = list()
 /atom/proc/ratvar_act()
 	return
 
+//This proc is called on the location of an atom when the atom is Destroy()'d
+/atom/proc/handle_atom_del(atom/A)
```
Comment (tigercat2000) [inline code/game/atoms.dm]: Put a return after this line, it's an empty proc

**Comment by tigercat2000** [comment]
LGTM.

**Pull request merged**